diff --git a/contracts/0.4.24/StETH.sol b/contracts/0.4.24/StETH.sol
index b8c8d47f2..0094ebcf8 100644
--- a/contracts/0.4.24/StETH.sol
+++ b/contracts/0.4.24/StETH.sol
@@ -474,6 +474,11 @@ contract StETH is IERC20, Pausable {
         return TOTAL_SHARES_POSITION_LOW128.getLowUint128();
     }
 
+    // MUNGED BY CERTORA
+    function _setTotalShares(uint256 newTotalShares) internal {
+        TOTAL_SHARES_POSITION_LOW128.setLowUint128(uint128(newTotalShares));
+    }
+
     /**
      * @return the amount of shares owned by `_account`.
      */
@@ -522,7 +527,9 @@ contract StETH is IERC20, Pausable {
         newTotalShares = _getTotalShares().add(_sharesAmount);
         require(newTotalShares & UINT128_HIGH_MASK == 0, "SHARES_OVERFLOW");
 
-        TOTAL_SHARES_POSITION_LOW128.setLowUint128(newTotalShares);
+        // MUNGED BY CERTORA
+        // TOTAL_SHARES_POSITION_LOW128.setLowUint128(newTotalShares);
+        _setTotalShares(newTotalShares);
 
         shares[_recipient] = shares[_recipient].add(_sharesAmount);
 
@@ -551,7 +558,9 @@ contract StETH is IERC20, Pausable {
         require(_sharesAmount <= accountShares, "BALANCE_EXCEEDED");
 
         newTotalShares = _getTotalShares().sub(_sharesAmount);
-        TOTAL_SHARES_POSITION_LOW128.setLowUint128(newTotalShares);
+        // MUNGED BY CERTORA
+        // TOTAL_SHARES_POSITION_LOW128.setLowUint128(newTotalShares);
+        _setTotalShares(newTotalShares);
 
         shares[_account] = accountShares.sub(_sharesAmount);
     }
