diff --git a/contracts/common/lib/MinFirstAllocationStrategy.sol b/contracts/common/lib/MinFirstAllocationStrategy.sol
index 606b35d48..2865d8b9b 100644
--- a/contracts/common/lib/MinFirstAllocationStrategy.sol
+++ b/contracts/common/lib/MinFirstAllocationStrategy.sol
@@ -31,7 +31,7 @@ library MinFirstAllocationStrategy {
         uint256[] memory buckets,
         uint256[] memory capacities,
         uint256 allocationSize
-    ) public pure returns (uint256 allocated, uint256[] memory) {
+    ) internal pure returns (uint256 allocated, uint256[] memory) { // MUNGED BY CERTORA
         uint256 allocatedToBestCandidate = 0;
         while (allocated < allocationSize) {
             allocatedToBestCandidate = allocateToBestCandidate(buckets, capacities, allocationSize - allocated);
